hikari/events/message_events.py

```diff
@@ -367,16 +402,29 @@ def guild_id(self) -> snowflakes.Snowflake:
         return guild_id
 
     @property
-    def author(self) -> guilds.Member:
+    def author(self) -> users.User:
         """Member that sent the message.
 
+        !!! note
+            For webhooks, this will be a `hikari.users.User`.
+
+            Any code relying on this being a `hikari.guilds.Member` directly
+            should use an `isinstance` assertion to determine if member info
+            is available or not.
+
         Returns
         -------
-        hikari.guilds.Member
-            The member that sent the message. This is a specialised
+        hikari.users.User
+            The member that sent the message, if known. This is a specialised
             implementation of `hikari.users.User`.
+
+            If the author was a webhook, then a `hikari.users.User` will be
+            returned instead, as webhooks do not have member objects.
         """
-        return typing.cast(guilds.Member, self.message.member)
+        member = self.message.member
+        if member is not None:
+            return member
```

Comment:
is that assignment to "member" necessary here (mypy should let you narrow the type of a model attribute within the scope of a function)?

Reply:
partially do it to avoid any risk of raciness where something else could ever override that if a user uses a multithreaded application elsewhere. It isn't overly important, it is just slightly more efficient as it performs two less getattrs each time tbh.